department-of-veterans-affairs/caseflow
# SeanC/APPEALS-5034 Decided non-comp issues are not getting closed as decided

Resolves [Decided non-comp issues are not getting closed as decided](https://jira.devops.va.gov/browse/APPEALS-5034)

# Jira Description
For SC's (supplemental claims) and HLR's (higher level reviews), if the business line is not compensation or pension, the decision reviews go to what we call the "non-comp queue", a lightweight version of queue that allows users to add decisions to request issues.

When users are adding decisions, the decision issues are getting added, but the request issues are not being closed as "decided". This allows users to go back in and edit later, and do another action such as withdraw the issue.

See DecisionReviewTask.complete_with_payload! where decision issues are getting created, the task is getting closed, but the request issues are not getting closed. RequestIssue.close_decided_issue! should be called somewhere.

# Description of Changes
For SC's (supplemental claims) and HLR's (higher level reviews), that are in the non-comp queue, when a user adds decision to a decision issue. The request issues are now being closed as "decided" and the completed_by_id is set to the user_id of the user that added the decision.

## Acceptance Criteria
- [x] When a user submits a decision for a non-comp decision review (via the non-comp queue), it closes the decided request issues with the status of "decided"
- [x] All other behavior is retained

## Testing Plan
1. Logged in as a user with non-comp claims(SC's and HLR's) in their decision review queue (ex. ACBAUERVVHAH)
2. Navigate to the decision review queue
3. Click on the first Supplemental Claim from the "In Progress tasks" tab
4. Save the task id for the claim somewhere (the last 4 digits in the claims url, ex. decision_reviews/vha/tasks/9779)
5. From the "Disposition" dropdown, select any of the options
6. Click on the decision date field and choose any date
7. Click on the "Complete" button
8. Verify that the claim is now in the "Completed tasks" tab
9. Open up the rails console and enter this command, but replace <task_id> with previously saved task_id `task = Task.find(<task_id>)`
<img width="1008" alt="Test_Instructions_ScreenShot1" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110493538/9b81025f-00cc-47d2-a954-7a8272c0ec56">

10. Note that the completed_by_id field is id of the user that just completed the disposition. (ex. ACBAUERVVHAH's id=2000006012) Command for getting user_id: `User.find_by(css_id: <user's css_id>)`
<img width="1132" alt="Test_Instructions_ScreenShot2" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110493538/36b13fbc-1115-4fca-9280-19dd0e3d3f9d">

11. Then run this command in the rails console `task.appeal.request_issues.first`
12. Note that `closed_status: "decided"`
<img width="1234" alt="Test_Instructions_ScreenShot3" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110493538/9b34ec80-5605-4213-a741-e3262a44e082">

13. Repeat steps 3-12 for a Higher Level Review

# Best practices
## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
